sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -103,6 +105,8 @@
    * @param name the displayed name for the new span.
    * @param kind the span kind.
    * @param parentSpanId the span_id of the parent span, or null if the new span is a root span.
+   * @param hasRemoteParent {@code true} if the parentContext is remote. {@code null} if this is a
+   *     root span.
```

Comment:
What about having this false for root Spans as well and use a primitive `boolean`

Reply:
Having it not primitive, `Boolean` helps to set it in a deterministic case; As that of a parentContext being remote and otherwise be null.
With changing it to primitive, we might have to cover all the places where it can be false?